src/pyodide/internal/snapshot.ts

```diff
@@ -296,9 +296,8 @@ function checkLoadedSoFiles(dsoJSON: DylinkInfo): void {
  * linear memory into MEMORY.
  */
 function makeLinearMemorySnapshot(
-  Module: Module
+  Module: Module, importedModulesList: string[]
 ): ArtifactBundler.MemorySnapshotResult {
-  const importedModulesList = memorySnapshotDoImports(Module);
   const dsoJSON = recordDsoHandles(Module);
   if (IS_CREATING_BASELINE_SNAPSHOT) {
```

Comment:
Is this `if` and `checkLoadedSoFiles` deprecated? should we remove them?

Reply:
Yes I think we should remove them. I originally wrote `checkLoadedSoFiles` when we didn't have proper test coverage, but now any relevant problems would cause a test failure even without the added assertions.

Reply:
I will delete this in a separate cleanup PR after this.